examples/dyn_create_destroy_apps/src/counter.rs

```diff
@@ -0,0 +1,77 @@
+use std::time::Duration;
+use yew::prelude::*;
+use yew_services::{
+    interval::{IntervalService, IntervalTask},
+    ConsoleService,
+};
+
+pub struct CounterModel {
+    counter: usize,
+    props: CounterProps,
+    _interval_task: IntervalTask,
+}
+
+#[derive(Clone, Properties)]
+pub struct CounterProps {
+    pub destroy_callback: Callback<()>,
+}
+
+pub enum CounterMessage {
+    Tick,
+}
+
+impl Component for CounterModel {
+    type Message = CounterMessage;
+
+    type Properties = CounterProps;
+
+    fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        // Create a Tick message every second
+        let interval_task = IntervalService::spawn(
+            Duration::from_secs(1),
+            link.callback(|()| Self::Message::Tick),
+        );
+        Self {
+            counter: 0,
+            props,
+            _interval_task: interval_task,
+        }
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        match msg {
+            // Count our internal state up by one
+            Self::Message::Tick => self.counter += 1,
+        }
+
+        true
```

Comment:
I prefer using this format because it helps avoid cases where components are needlessly updated.
```suggestion
        match msg {
            // Count our internal state up by one
            Self::Message::Tick => {
                self.counter += 1;
                true
            }
        }
```

Reply:
Fair enough